src/tools/script_tools.cpp

```diff
@@ -317,8 +327,18 @@ int RefreshScriptTools(ToolManager* toolMgr)
 	ParseDirAndCreateTools(toolMgr, QDir(":/scripts"), "Scripts", luaShell, true);
 	ParseDirAndCreateTools(toolMgr, app::UserScriptDir(), "Scripts", luaShell, true);
 
+	if(app::UserDataDir().exists("custom_user_script_dirs")){
```

Comment:
This looks very similar to the block above (as do the two lines above). Maybe there could be a separate function which does the parsing of standard and user tools? E.g.:

```
void ParseAndCreateTools (toolMgr, luaShell, false/true);
```

I hope the idea is clear. If not please tell me 😄